aten/src/ATen/Declarations.cwrap

```diff
@@ -3132,15 +3132,6 @@
         - CONSTANT AS_REAL(1)
         - THTensor* self
         - THTensor* mat2
-    - cname: spaddmm
```

Comment:
I don't really get this -- why does this not work for variables?

Reply:
`mm`'s signature is `mm(self, mat2)`. For sparse variables, we would need `mm(sparse, dense)` and it would dispatch on the sparse type. Right now we don't support having variable methods that dispatch on the sparse type that takes dense arguments (the other way around, having variable methods that dispatch on the dense type that take sparse arguments works with the `aten_dense_sparse` keyword).

To solve this, I could either add a `aten_sparse_dense` keyword or rewrite `mm` as a native function and have it call `addmm` when `self` is sparse.

Reply:
ok, I was confused here because when you say "variable methods" I think you mean "aten methods", i.e. this isn't some limitation of VariableType, this is a limitation of aten.  Do I have that right?